File: backend/src/main/java/com/back/domain/product/product/entity/Product.java
package com.back.domain.product.product.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import lombok.*;

import static jakarta.persistence.GenerationType.IDENTITY;

@Entity
@Getter
@Setter
@ToString
@NoArgsConstructor
public class Product {
    @Id
    @GeneratedValue(strategy = IDENTITY)
    private long id;
    private String name;
    private int price;
    @Column(columnDefinition = "TEXT")
    private String description;
    private String imageUrl;

    public Product(String name, int price, String description, String imageUrl) {
        this.name = name;
        this.price = price;
        this.description = description;
        this.imageUrl = imageUrl;
    }

    public void modify(String name, int price, String description, String imageUrl) {
        this.name = name;
        this.price = price;
        this.description = description;
        this.imageUrl = imageUrl;
    }
}

File: frontend/src/types/product.ts
export type Product = {
  id: number;
  name: string;
  price: number;
  description: string;
  imageUrl: string;
};

export type ProductRequest = {
  name: string;
  price: number;
  description: string;
  imageUrl: string;
};

export type ProductSale = {
  id: number;
  sales: number;
};
